### PR TITLE
MOB-78 Send Welcome Messages Alternative

### PR DESCRIPTION
**Pull request by BenjaminSsempala**
**_WHAT DOES THIS PR DO?_**
- Send welcome mesages (Alternative to kafka)

**_WHAT ISSUES ARE RELATED TO THIS PR?_**

- Jira cards
    - [MOB-78](https://airqoteam.atlassian.net/browse/MOB-78)
- GitHub issues
    - Closes #<enter GitHub issue number here>

**_HOW DO I TEST OUT THIS PR?_**
- Checkout this branch and navigate to the `firebase/auth/functions` directory.
- Run` npm init`
- Run `firebase emulators:start`
- You can proceed with the testing.

**_WHICH ENDPOINTS SHOULD BE READY FOR TESTING?:_** 

**_IS THERE ANY JENKINS CONSOLE LINK FOR CODE COVERAGE AND BUILD INFO?_**

**_ARE THERE ANY RELATED PRs?_**

[MOB-78]: https://airqoteam.atlassian.net/browse/MOB-78?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/firebase/auth/config/emailTemplates.js [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+    <p> With the AirQo app, you'll have access to:<p/>
+    <p>1. Air quality analytics - view air quality readings by day/week in different locations</p>
+    <p>2. For You - personalized air quality recommendations based on what you share frequently and your favorite locations</p>
+    <p>3. Search - find locations by location name or by navigating the map</p>
+    <p>4. Know your air - a fun way of learning about air quality</p>
+    <p>We've designed it to be easy to use and navigate, so you can find what you're looking for quickly. Get air quality information like air quality lessons and tips on how to reduce air pollution that you can share with your pals through text or visual updates.</p>
+    <p>We're constantly updating and improving our app to make sure you have the best experience possible. If you have any questions or feedback, please don't hesitate to reach out to us through the app's support feature</p>
+    <p>Thank you for choosing our app, and we can't wait for you to see what it can do for you. Happy exploring!</p>`;
+  },
+};
```
Comment (BenjaminSsempala) [inline src/firebase/auth/config/emailTemplates.js:15]: Welcome Message Template

File: src/firebase/auth/functions/index.js [hunk 2]
```diff
@@ -22,37 +34,30 @@ async function sendGoodByeMessage(_user) {
...
+  functions.logger.log("New welcome email sent to:", email);
+  return null;
+}
+
+exports.sendWelcomeEmail = functions.auth.user().onCreate((user) => {
+  if (user.email !== null) {
+    const email = user.email;
+    return sendWelcomeEmail(email);
+  }
+});
```
Comment (BenjaminSsempala) [inline src/firebase/auth/functions/index.js]: Send Email Function

File: src/firebase/auth/functions/index.js [hunk 3]
```diff
@@ -145,7 +150,7 @@ async function checkIfUserExists(data) {
 
 exports.onUserSignUp = functions.auth.user().onCreate(async (user) => {
   if (user.email !== null) {
-    return await produceMessage(user);
+    // return await produceMessage(user);
```
Comment (BenjaminSsempala) [inline src/firebase/auth/functions/index.js:153]: Commenting out the Kafka Produce message until kafka is configured @NoahNsimbe

File: src/firebase/auth/functions/index.js [hunk 4]
```diff
@@ -22,37 +34,23 @@ async function sendGoodByeMessage(_user) {
...
+    },
+    to: email,
+    subject: "Welcome to AirQo!",
+    html: emailTemplate.mobileAppWelcome(),
+  };
+
+  await transporter.sendMail(mailOptions);
+  functions.logger.log("New welcome email sent to:", email);
+  return null;
+}
```
Comment (BenjaminSsempala) [inline src/firebase/auth/functions/index.js:53]: Send welcome message

**Review by Baalmart — APPROVED** [review]
This looks good to me. Thanks @BenjaminSsempala !

**Comment by Baalmart** [comment]
@BenjaminSsempala , please share some screenshots for your successful tests.
I have just reviewed the code which looks good from my end. 

I just want to "visually" run the PR. Hence my request for the screenshots. :) 

Thanks again!

File: src/firebase/auth/functions/index.js (hunk 3 above)
Comment (NoahNsimbe) [reply src/firebase/auth/functions/index.js:153]: Thanks @BenjaminSsempala

**Review by NoahNsimbe — APPROVED** [review]
LGTM

**Comment by BenjaminSsempala** [comment]
> @BenjaminSsempala , please share some screenshots for your successful tests. I have just reviewed the code which looks good from my end.


### @Baalmart I tested using the firebase emulator.

image

### This ran succssfully, so I hope you received an email.
image

### Email comes like this
image

**Comment by Baalmart** [comment]
Thanks @BenjaminSsempala brilliant! 👏🏾👏🏾

**Pull request merged**